pkg/controller/common/container/container_test.go

```diff
@@ -66,6 +53,92 @@ func TestImageRepository(t *testing.T) {
 			suffix:     "-obi1",
 			want:       testRegistry + "/elastic/elasticsearch-obi1:42.0.0",
 		},
+		{
+			name:       "Elasticsearch 8 image in ubi mode",
+			image:      ElasticsearchImage,
+			version:    "8.12.0",
+			repository: "elastic",
+			suffix:     "-ubi",
+			want:       testRegistry + "/elastic/elasticsearch-ubi:8.12.0",
+		},
+		{
+			name:    "Elasticsearch old 8 image in ubi mode uses old -ubi8 suffix",
+			image:   ElasticsearchImage,
+			version: "8.11.0",
+			suffix:  "-ubi",
+			want:    testRegistry + "/elasticsearch/elasticsearch-ubi8:8.11.0",
+		},
+		{
+			name:       "Elasticsearch 7 image in ubi mode",
+			image:      ElasticsearchImage,
+			version:    "7.17.16",
+			repository: "elastic",
+			suffix:     "-ubi",
+			want:       testRegistry + "/elastic/elasticsearch-ubi:7.17.16",
+		},
+		{
+			name:    "Elasticsearch old 7 image in ubi mode uses old -ubi8 suffix",
+			image:   ElasticsearchImage,
+			version: "7.17.15",
+			suffix:  "-ubi",
+			want:    testRegistry + "/elasticsearch/elasticsearch-ubi8:7.17.15",
+		},
+		{
+			name:       "Maps 7 image with custom repository always uses -ubi suffix",
```

Comment:
Do we know that maps is going to adopt the new convention? (If not we should push for it for consistency)